### PR TITLE
fix: Removing warnings and organizing build scripts

### LINKED ISSUE #525: Backend - Fix build scripts
1. Guarantee support for many versions of dependency manager and JDK, maintaining Java 8 target.
2. Clean up build warnings.

### PR DESCRIPTION
**Pull request by ze12augusto**
Signed-off-by: José Augusto <ze12augusto@gmail.com>

## Description

* Removing warnings and organizing build scripts*

## Related Issues

*Fix #525  *

## Checklist

Before you create this PR confirms that it meets all requirements listed below by checking the relevant checkboxes (`[x]`). This will ensure a smooth and quick review process.

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I signed the [DCO].
- [x] All existing and new tests are passing.
- [x] I am willing to follow-up on review comments in a timely manner.

## Breaking Change

Did any tests fail when you ran them? Please read [Handling breaking changes].

- [x] No, no existing tests failed, so this is *not* a breaking change.

[issue database]: https://github.com/ZupIT/beagle/issues
[DCO]: https://developercertificate.org/

### CODE REVIEW

File: backend/processor/build.gradle [hunk 1]
```diff
@@ -39,10 +36,24 @@ dependencies {
     kapt Dependencies.ProcessorLibraries.incapPrcessor
     kaptTest project(Dependencies.Modules.processor)
 
-    testRuntime Dependencies.TestLibraries.junit5Engine
+    runtimeOnly Dependencies.TestLibraries.junit5Engine
```
Comment (LG95) [inline backend/processor/build.gradle:39]: Why this change?

File: backend/processor/build.gradle [hunk 2]
```diff
@@ -39,10 +36,24 @@ dependencies {
...
+        javaParameters = true
+    }
+}
+
+compileTestKotlin {
+    kotlinOptions {
+        jvmTarget = Dependencies.Versions.java
+        javaParameters = true
+    }
+}
```
Comment (LG95) [inline backend/processor/build.gradle:57]: Couldn't this be configured in the backend root script?

File: backend/build.gradle [hunk 3]
```diff
@@ -16,7 +16,6 @@
 
 buildscript {
     repositories {
-        maven { url "https://plugins.gradle.org/m2/" }
         google()
```
Comment (LG95) [inline backend/build.gradle:19]: What dependency of ours comes from this repo?

File: backend/processor/build.gradle (hunk 1 above)
Comment (ze12augusto) [reply backend/processor/build.gradle:39]: Its deprecated

Comment (LG95) [reply backend/processor/build.gradle:39]:
```suggestion
    testRuntimeOnly Dependencies.TestLibraries.junit5Engine
```

File: backend/sample/core/build.gradle [hunk 4]
```diff
@@ -16,24 +16,12 @@
 
 import br.com.zup.beagle.Dependencies
 
-apply plugin: 'kotlin'
 apply plugin: 'kotlin-kapt'
```
Comment (LG95) [inline backend/sample/core/build.gradle:19]:
```suggestion
```

File: backend/sample/core/build.gradle [hunk 5]
```diff
@@ -16,24 +16,12 @@
...
 
-group = Dependencies.GroupId.backendSample
-version = Dependencies.Releases.beagleVersionName
-sourceCompatibility = JavaVersion.VERSION_1_8
-
 sourceSets {
     main.java.srcDirs += 'build/generated/source/kapt'
 }
```
Comment (LG95) [inline backend/sample/core/build.gradle:23]:
```suggestion
```

File: backend/sample/micronaut/build.gradle [hunk 6]
```diff
@@ -30,31 +29,18 @@ configurations {
...
-        javaParameters = true
-    }
-}
-
-shadowJar {
-    archiveClassifier = ""
-    mergeServiceFiles()
-}
-
 run.classpath += configurations.developmentOnly
```
Comment (LG95) [inline backend/sample/micronaut/build.gradle:45]:
```suggestion

tasks.withType(JavaCompile) {
    options.compilerArgs = [
        '-Amicronaut.processing.incremental=true'
    ]
}

```
